examples/threading/threading.c

```diff
@@ -14,6 +14,39 @@ void* threadfunc(void* thread_param)
     // TODO: wait, obtain mutex, wait, release mutex as described by thread_data structure
     // hint: use a cast like the one below to obtain thread arguments from your parameter
     //struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+    struct thread_data* threadfunc_args = (struct thread_data *) thread_param;
+    
+    threadfunc_args-> thread_complete_success = false;
+
+    //wait to lock mutex
+    int rc = usleep(threadfunc_args->wait_to_start_mutex); 
+    if( rc != 0){
+        printf("wait for mutex failed!!!!!\n"); 
+        return thread_param;
+    }
+
+    //Lock mutex
+    rc = pthread_mutex_lock(threadfunc_args->mutex);
+    if( rc != 0){
+        printf("mutex start failed!!!!!\n"); 
+        return thread_param;
+    }
+
+    //Wait to release mutex
+    rc = usleep(threadfunc_args->wait_to_release_mutex);
+    if( rc != 0){
+        printf("wait for mutec release failed!!!!!\n"); 
+        return thread_param;
```

Comment:
@ajsanthosh14 not safe to return here since you are holding the locked mutex.

Reply:
@Maitreyee2095  I totally agree! honestly, I didn't consider the scenario.